src/ext/respToRedisLoader.c

```diff
@@ -400,45 +477,329 @@ _LIBRDB_API RdbxRespToRedisLoader *RDBX_createRespToRedisFd(RdbParser *p,
-    if (sockfd == -1) {
-        RDB_reportError(p, (RdbRes) RDBX_ERR_RESP2REDIS_CREATE_SOCKET, "Failed to create TCP socket");
+                                                             int port,
+                                                             RdbxSSLConfig *sslConfig) {
+    int sockfd = -1;
+    int origSockFlags = 0;
+    RdbxRespToRedisLoader *ctx = NULL;
+
+    /* Step 1: Create TCP connection */
+    sockfd = createTcpConnection(p, hostname, port);
+    if (sockfd == -1)
         return NULL;
-    }
 
-    struct sockaddr_in server_addr;
-    memset(&server_addr, 0, sizeof(server_addr));
-    server_addr.sin_family = AF_INET;
-    server_addr.sin_port = htons(port);
-    if (inet_pton(AF_INET, hostname, &(server_addr.sin_addr)) <= 0) {
-        RDB_reportError(p, (RdbRes) RDBX_ERR_RESP2REDIS_INVALID_ADDRESS,
-                        "Failed to convert IP address. inet_pton(hostname=%s, port=%d) => errno=%d",
-                        hostname, port, errno);
-        goto createErr;
+    /* Step 2: Configure socket options */
+    if (configureSocket(p, sockfd, &origSockFlags) != 0) {
+        close(sockfd);
+        return NULL;
     }
 
-    if (connect(sockfd, (struct sockaddr *) &server_addr, sizeof(server_addr)) == -1) {
-        RDB_reportError(p, (RdbRes) RDBX_ERR_RESP2REDIS_FAILED_CONNECT,
-                        "Failed to connect(hostname=%s, port=%d) => errno=%d",
-                        hostname, port, errno);
-        goto createErr;
+    /* Step 3: Allocate and initialize loader context */
+    ctx = RDB_alloc(p, sizeof(RdbxRespToRedisLoader));
+    if (!ctx) {
+        RDB_reportError(p, (RdbRes) RDBX_ERR_RESP_FAILED_ALLOC,
+                        "Failed to allocate struct RdbxRespToRedisLoader");
+        close(sockfd);
+        return NULL;
     }
 
-    RdbxRespToRedisLoader *res = RDBX_createRespToRedisFd(p, rdbToResp, auth, sockfd);
+    initLoaderContext(ctx, p, sockfd, origSockFlags);
 
-    if (!res) goto createErr;
+#ifdef USE_OPENSSL
+    /* Step 4: Setup SSL/TLS if requested */
+    if (sslConfig != NULL) {
+        if (createSslContext(p, ctx, sslConfig) != 0)
+            goto error_cleanup;
 
-    res->fdOwner = 1;
-    return res;
+        if (performSslHandshake(p, ctx, sslConfig, sockfd) != 0)
+            goto error_cleanup;
+    }
+#else
+    /* SSL not compiled in */
+    if (sslConfig != NULL) {
+        RDB_reportError(p, (RdbRes) RDBX_ERR_SSL_INIT_FAILED,
+                        "SSL/TLS support not compiled in. Rebuild with BUILD_TLS=yes");
+        goto error_cleanup;
+    }
+#endif
+
+    /* Step 5: Authenticate if needed */
+    if (auth && (redisAuth(ctx, auth) != RDB_OK)) {
+        RDB_reportError(p, (RdbRes) RDBX_ERR_RESP2REDIS_AUTH_FAILED,
+                        "Redis authentication failed.");
+        goto error_cleanup;
+    }
+
+    /* Step 6: Attach writer and return */
+    RdbxRespWriter inst = {ctx, redisLoaderDestroy, redisLoaderWritev, redisLoaderFlush};
```

Comment:
Who is going to `close(sockfd)` later on? Will openssl handle it as we attached socket to ssl context via `ssl_set_fd()`?